### DISCUSSION TITLE
How to change the Port of the integration without removing / and reinstalling it from HA

### DISCUSSION THREAD

**Discussion by ChristophCaina** [binsentsu/home-assistant-solaredge-modbus#118] [Q&A]
It might happen, that you setup your configuration to listen to Port 1502 on the inverter.
The integration is working - but after some changes on your setup, it stops and all values become "unknown".

SolarEdge Inverters are using the ports 1502 OR 502 by default for connections over Modbus - and it might be, that the connected port does change.

However, there isn't a way to reconfigure your integration from the UI... therefore, you cannot change the port "easily"...

The following instruction will explain, how to change ANY settings from the integration - without uninstalling it.
This can either be changing the port, or changing settings for the amount of meters, etc.

1. you need to open the following file from HomeAssistant:
`/config/.storage/core.config_entries`

this will provide you a list of all installed integrations as a JSON-File.

2. search for your SolarEdge Modbus integration:
```
      {
        "entry_id": "58dbc7a51f29eb233550bb8cc2c7cc70",
        "version": 1,
        "domain": "solaredge_modbus",
        "title": "solaredge",
        "data": {
          "name": "solaredge",
          "port": 502,
          "read_meter_1": true,
          "read_meter_2": false,
          "read_meter_3": false,
          "read_battery_1": false,
          "read_battery_2": false,
          "scan_interval": 30,
          "host": "192.168.xxx.xxx"
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": "192.168.xxx.xxx",
        "disabled_by": null
      },
```

In this file, you can now change the IP, the Port or you can enable / disable the "read_meter" or "read_battery" settings.
After the changes, it might be required to reload the integration or maybe, to restart HomeAssistant...

But it should not be required to uninstall and install the integration again.

**Comment by Centurytom** [comment]
Hi, thanks for that post but in my case it won´t work. After I am setting the new IP which I want to change and restart HA... the IP switched back to its previous one.... ? I tried to change that settings via SSH with VI .... and with StudioCode integration. Nothing worked for me.

**Comment by ChristophCaina** [comment]
Hi,
#368 should solve this issue and provide a "reconfigure" option for the integration
